### PR TITLE
Dockerize app for code level metrics

### PR DESCRIPTION
**Pull request by bizob2828**
This PR does two things:
 * Updates the example app to leverage `getServerSideProps`, middleware, and tie the front end and back end together via http calls to the api.
   * This was done to show all the capabilities of the New Relic Node.js Next.js instrumentation.  
   * I also upgraded to Next.js 12 to take advantage of middleware
 * Dockerizes the Next.js app to help facilitate integration testing with [CodeStream code level metrics](https://newrelic.com/blog/nerdlog/codestream-code-level-metrics)
 * See `README_CLM.md` for how to use the docker container.

### CODE REVIEW

**Review by matewilk — COMMENTED** [review]
Hi @bizob2828 ,

Thanks for contributing.
I'll test it out as soon as I can.

File: pages/_document.tsx [hunk 1]
```diff
@@ -20,9 +20,6 @@ class MyDocument extends Document {
       hasToRemoveScriptWrapper: true,
     });
 
-    newrelic.setTransactionName("App started transaction");
```
Comment (matewilk) [inline pages/_document.tsx:23]: These two lines are linked to/mentioned in the public blog post

File: pages/_document.tsx [hunk 2]
```diff
@@ -20,9 +20,6 @@ class MyDocument extends Document {
...
-    newrelic.addCustomAttribute("AppAttribute", "NextJs New Relic attribute");
```
Comment (matewilk) [inline pages/_document.tsx:24]: These two lines are linked to/mentioned in the public blog post

Comment (bizob2828) [reply pages/_document.tsx:24]: I would update blog post. the Next.js instrumentation names transactions

Comment (bizob2828) [reply pages/_document.tsx:24]: If you keep this all the instrumentation naming gets superseded by this

File: pages/posts/[id].tsx [hunk 3]
```diff
@@ -16,53 +16,30 @@ function Post({ post }) {
...
+  });
+
+  const [ post ] = posts.filter((p) => p.id === parseInt(id, 10))
+
+  return {
+    props: {
+      post,
+    },
   };
-  logger.info("Getting post id", { postId: params.id });
```
[the remark this answers is not in the document]
Comment (bizob2828) [reply pages/posts/[id].tsx:58]: ah ok

**Pull request merged**